src/main/groovy/de/dkfz/roddy/execution/jobs/cluster/slurm/SlurmJobManager.groovy

```diff
@@ -138,136 +149,174 @@ class SlurmJobManager extends GridEngineBasedJobManager {
     /**
      * Reads the scontrol output and creates GenericJobInfo objects
      * @param resultLines - Input of ExecutionResult object
-     * @return map with jobid as key
+     * @return GenericJobInfo
      */
-    protected Map<BEJobID, GenericJobInfo> processExtendedOutput(String stdout, Map<BEJobID, GenericJobInfo> result) {
+    protected GenericJobInfo processExtendedOutput(String stdout) {
         // Create a complex line object which will be used for further parsing.
         ComplexLine line = new ComplexLine(stdout)
 
         Collection<String> splitted = line.splitBy(" ").findAll { it }
-        if (splitted.size() > 1) {
-            Map<String, String> jobResult = [:]
+        if (splitted.size() <= 1) {
+            return null
+        }
 
-            for (int i = 0; i < splitted.size(); i++) {
-                String[] jobKeyValue = splitted[i].split("=")
-                if (jobKeyValue.size() > 1) {
-                    jobResult.put(jobKeyValue[0], jobKeyValue[1])
+        Map<String, String> jobResult = [:]
+
+        for (int i = 0; i < splitted.size(); i++) {
```

Comment:
Typical Roddy/BE spaghetti code. Consider extracting into a function with a descriptive name.